src/TypeScript/Rules/ActiveJavaScriptRulesProvider.cs

```diff
@@ -39,28 +40,42 @@ interface IActiveJavaScriptRulesProvider
     internal class ActiveJavaScriptRulesProvider : IActiveJavaScriptRulesProvider
     {
         private readonly IJavaScriptRuleDefinitionsProvider jsRuleDefinitions;
+        private readonly IUserSettingsProvider userSettingsProvider;
 
         [ImportingConstructor]
-        public ActiveJavaScriptRulesProvider(IJavaScriptRuleDefinitionsProvider jsRuleDefinitions)
+        public ActiveJavaScriptRulesProvider(IJavaScriptRuleDefinitionsProvider jsRuleDefinitions,
+            IUserSettingsProvider userSettingsProvider)
         {
             this.jsRuleDefinitions = jsRuleDefinitions;
+            this.userSettingsProvider = userSettingsProvider;
```

Comment:
Note: there is currently no caching i.e. the set of active rules is recalculated each time.

I'll add a separate ticket for this, but I think we can ship the first version without it.
FYI the CFamily analyzer has a cache, but it only works in standalone mode,; the active rules are recalculated each time in connected mode, which involves a much more disc access than this class.